### PR TITLE
A fix to allow leading zero on sig values

### PR DESCRIPTION
**Pull request by poplexity**

### CODE REVIEW

File: translator/src/transaction.rs [hunk 1]
```diff
@@ -96,7 +106,7 @@ impl TelosEVMTransaction {
         action: TransferAction,
         native_to_evm_cache: &NameToAddressCache,
     ) -> Self {
-        let address: Address = if action.memo.starts_with("0x") {
+        let address: Address = if action.memo.len() == 42 && action.memo.starts_with("0x") {
```
Comment (lesa-telos) [inline translator/src/transaction.rs:109]: 42 might be good to extract to a constant

File: translator/src/transaction.rs [hunk 2]
```diff
@@ -46,16 +46,26 @@ impl TelosEVMTransaction {
         let tx_raw = &mut raw.tx.as_slice();
 
         if tx_raw[0] >= 0xc0 && tx_raw[0] <= 0xfe {
-            let signed_legacy_result = TxLegacy::decode_signed_fields(tx_raw);
+            let mut signed_legacy_result = TxLegacy::decode_signed_fields(tx_raw);
+            // If we fail to decode with the strict RLP from reth,
+            // and we don't have a raw.sender which suggests a native signed trx
+            // then try the telos legacy decode without passing a signature
+            if signed_legacy_result.is_err() && raw.sender.is_none() {
```
Comment (lesa-telos) [inline translator/src/transaction.rs:53]: This should be the same without having 2 if checks for signed_legacy_result error
```suggestion
if signed_legacy_result.is_err() {
            if raw.sender.is_none() {
                signed_legacy_result =
                    TxLegacy::decode_telos_signed_fields(&mut raw.tx.clone().as_slice(), None);
            }
            let address = Address::from(
                raw.sender
                    .expect("Failed to get address from sender in unsigned transaction, signed_legacy_result is Err")
                    .data,
            );
            let sig = make_unique_vrs(block_hash, address, trx_index);
            let unsigned_legacy = TxLegacy::decode_telos_signed_fields(
                &mut raw.tx.clone().as_slice(),
                Some(sig),
            )?;
            let envelope = TxEnvelope::Legacy(unsigned_legacy);
            return Ok(TelosEVMTransaction { envelope, receipt });
        }
```

**Review by lesa-telos — APPROVED** [review]
2 minor comments, lgtm

File: translator/src/rlp/telos_rlp_decode.rs [hunk 3]
```diff
@@ -45,21 +51,36 @@ impl TelosTxDecodable for TxLegacy {
...
+                    let decoded_signature = Signature::decode_rlp_vrs(buf)?;
+                    v = decoded_signature.v();
+                    r = decoded_signature.r();
+                    s = decoded_signature.s();
+                }
+            }
+            provided_sig.unwrap()
+        };
+
+        if provided_sig.is_some() && (v.to_u64() != 0 || r != U256::ZERO || s != U256::ZERO) {
```
Comment (coa-telos) [inline translator/src/rlp/telos_rlp_decode.rs:83]: Also this means that this check applies only to `Some(sig) => ` branch and can be moved there?

**Review by coa-telos — APPROVED** [review]
Looks fine, just minor comments.

File: translator/src/transaction.rs (hunk 2 above)
Comment (poplexity) [reply translator/src/transaction.rs:53]: This does change the behavior, if we decode_telos_signed_fields with a None signature successfully then we don't want to enter the second if block.

File: translator/src/rlp/telos_rlp_decode.rs (hunk 3 above)
Comment (poplexity) [reply translator/src/rlp/telos_rlp_decode.rs]: Good catch, made this change.

File: translator/src/rlp/telos_rlp_decode.rs [hunk 4]
```diff
@@ -45,21 +51,36 @@ impl TelosTxDecodable for TxLegacy {
...
-            //   in RLP encoding these 0 values are encoded as [128, 128, 128], so we need purge them
-            //   from the buffer but leave signature value as zeros
-            if buf == &[128, 128, 128] {
-                buf.advance(3);
-            } else {
-                let signature = Signature::decode_rlp_vrs(buf)?;
-                v = signature.v();
-                r = signature.r();
-                s = signature.s();
+        let sig = if provided_sig.is_none() {
```
[the remark this answers is not in the document]
Comment (poplexity) [reply translator/src/rlp/telos_rlp_decode.rs]: I changed the if to a match, is that what you meant?

**Pull request merged**